rust/src/libnm_dbus/connection/conn.rs

```diff
@@ -65,35 +63,48 @@ impl TryFrom<NmConnectionDbusOwnedValue> for NmConnection {
-        }
-        if let Some(ipv4_set) = value.get("ipv4") {
-            nm_con.ipv4 = Some(NmSettingIp::try_from(ipv4_set)?);
-        }
-        if let Some(ipv6_set) = value.get("ipv6") {
-            nm_con.ipv6 = Some(NmSettingIp::try_from(ipv6_set)?);
-        }
-        if let Some(br_value) = value.get("bridge") {
-            nm_con.bridge = Some(NmSettingBridge::try_from(br_value)?);
-        }
-        if let Some(br_port_value) = value.get("bridge-port") {
-            nm_con.bridge_port =
-                Some(NmSettingBridgePort::try_from(br_port_value)?);
-        }
-        if let Some(ovs_br_value) = value.get("ovs-bridge") {
-            nm_con.ovs_bridge =
-                Some(NmSettingOvsBridge::try_from(ovs_br_value)?);
-        }
-        if let Some(ovs_port_value) = value.get("ovs-port") {
-            nm_con.ovs_port = Some(NmSettingOvsPort::try_from(ovs_port_value)?);
-        }
-        if let Some(ovs_iface_value) = value.get("ovs-interface") {
-            nm_con.ovs_iface =
-                Some(NmSettingOvsIface::try_from(ovs_iface_value)?);
-        }
-        if let Some(wired_value) = value.get("802-3-ethernet") {
-            nm_con.wired = Some(NmSettingWired::try_from(wired_value)?);
+        for (key, setting_value) in value.iter() {
+            match key.as_str() {
+                "connection" => {
+                    nm_con.connection =
+                        Some(NmSettingConnection::try_from(setting_value)?);
+                }
+                "ipv4" => {
+                    nm_con.ipv4 = Some(NmSettingIp::try_from(setting_value)?);
+                }
+                "ipv6" => {
+                    nm_con.ipv6 = Some(NmSettingIp::try_from(setting_value)?);
+                }
+                "bridge" => {
+                    nm_con.bridge =
+                        Some(NmSettingBridge::try_from(setting_value)?);
+                }
+                "bridge-port" => {
+                    nm_con.bridge_port =
+                        Some(NmSettingBridgePort::try_from(setting_value)?);
+                }
+                "ovs-bridge" => {
+                    nm_con.ovs_bridge =
+                        Some(NmSettingOvsBridge::try_from(setting_value)?);
+                }
+                "ovs-port" => {
+                    nm_con.ovs_port =
+                        Some(NmSettingOvsPort::try_from(setting_value)?);
+                }
+                "ovs-interface" => {
+                    nm_con.ovs_iface =
+                        Some(NmSettingOvsIface::try_from(setting_value)?);
+                }
+                "802-3-ethernet" => {
+                    nm_con.wired =
+                        Some(NmSettingWired::try_from(setting_value)?);
+                }
+                _ => {
+                    nm_con
+                        ._other
+                        .insert(key.to_string(), setting_value.clone());
+                }
+            }
         }
```

Comment:
Looking at it, we could avoid most of the cloning by passing it from here a owned value. If you don't mind I can create a commit on top of this one to show what I mean.